src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Lists/BitBasicListTests.cs

```diff
@@ -0,0 +1,134 @@
+
+namespace Bit.Client.Web.BlazorUI.Tests.Lists
+{
+    [TestClass]
+    public class BitBasicListTests : BunitTestContext
+    {
+
+        [DataTestMethod,
+            DataRow(Visual.Fluent, true, 1000),
+            DataRow(Visual.Fluent, false, 100),
+
+            DataRow(Visual.Cupertino, true, 1000),
+            DataRow(Visual.Cupertino, false, 100),
+
+            DataRow(Visual.Material, true, 1000),
+            DataRow(Visual.Material, false, 100),
+            ]
+        //When we use virtualization, two additional parts are created,
+        //that's why this method use itemCount +2
+        public void BitBasicListShoudRenderExpectedChildElements(Visual visual, bool virtualize, int itemCount)
+        {
+            var component = RenderComponent<BitBasicListTest>(parameters =>
+            {
+                parameters.Add(p => p.Visual, visual);
+                parameters.Add(p => p.Virtualize, virtualize);
+                parameters.Add(p => p.Items, GetTestData(itemCount));
+            });
+
+            var bitList = component.Find(".bit-bsc-lst");
+
+            var visualClass = visual == Visual.Cupertino ? "cupertino" : visual == Visual.Material ? "material" : "fluent";
+            Assert.IsTrue(bitList.ClassList.Contains($"bit-bsc-lst-{visualClass}"));
+
+            int comItemCount = bitList.ChildElementCount;
+            if (virtualize)
+            {
+                Assert.AreEqual(comItemCount, itemCount + 2);
+            }
+            else
+            {
+                Assert.AreEqual(comItemCount, itemCount);
+            }
+        }
+
+        [DataTestMethod,
+            DataRow(Visual.Fluent, 1000, true, 1),
+            DataRow(Visual.Fluent, 100, false, 1),
+            
+            DataRow(Visual.Cupertino, 1000, true, 1),
+            DataRow(Visual.Cupertino, 100, false, 1),
+            
+            DataRow(Visual.Material, 1000, true, 1),
+            DataRow(Visual.Material, 100, false, 1),
+            ]
+        public void BitBasicListRenderExtraItemsUsingOverScanCount(Visual visual,
+             int itemCount, bool virtualize, int overScanCount)
+        {
+            var component = RenderComponent<BitBasicListTest>(parameters =>
+            {
+                parameters.Add(p => p.Visual, visual);
+                parameters.Add(p => p.Virtualize, virtualize);
+                parameters.Add(p => p.Items, GetTestData(itemCount));
+                parameters.Add(p => p.OverscanCount, overScanCount);
+            });
+
+            var bitList = component.Find(".bit-bsc-lst");
+
+            var visualClass = visual == Visual.Cupertino ? "cupertino" : visual == Visual.Material ? "material" : "fluent";
+            Assert.IsTrue(bitList.ClassList.Contains($"bit-bsc-lst-{visualClass}"));
+
+            int comItemCount = bitList.ChildElementCount;
+            if (virtualize)
+            {
+                Assert.AreEqual(comItemCount, itemCount + (overScanCount + 2));
```

Comment:
again, in virtualized mode, the number of elements rendered is about the size of a single page not about the size of all of the data

Reply:
and one more thing, the default value of OverScanCount in the Blazor Virtualize component is 3, so this value is should be considered in the other tests.